pqEventTranslator.cxx

```diff
@@ -39,40 +40,109 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "pqComboBoxEventTranslator.h"
 #include "pqDoubleSpinBoxEventTranslator.h"
 #include "pqEventComment.h"
+#include "pqEventTypes.h"
 #include "pqLineEditEventTranslator.h"
+#include "pqListViewEventTranslator.h"
 #include "pqMenuEventTranslator.h"
+#include "pqNativeFileDialogEventTranslator.h"
 #include "pqObjectNaming.h"
 #include "pqSpinBoxEventTranslator.h"
 #include "pqTabBarEventTranslator.h"
+#include "pqTableViewEventTranslator.h"
 #include "pqTreeViewEventTranslator.h"
-#include "pqNativeFileDialogEventTranslator.h"
-#include "pq3DViewEventTranslator.h"
 
 #include <QCoreApplication>
 #include <QtDebug>
 #include <QSet>
 #include <QVector>
+#include <QPainter>
+#include <QApplication>
+#include <QToolBar>
+
+// Check Overlay class
+class pqCheckEventOverlay : public QWidget {
+public:
+  pqCheckEventOverlay(QWidget * parent = 0) : QWidget(parent) {
+    setAttribute(Qt::WA_NoSystemBackground);
+    setAttribute(Qt::WA_TransparentForMouseEvents);
+    this->Valid = false;
+    this->GlWidget = false;
+    this->Specific = false;
+    this->setObjectName("Overlay");
+  }
+
+  // true if the overlayed widget can be checked, false otherwise
+  bool Valid;
+
+  // true if the overlayed widget in an opengl widget, false otherwise
+  bool GlWidget;
+
+  // true if the size of the overlay is specifically defined
+  // via resize slot, false otherwise
+  bool Specific;
+
+  // Static overlauy margin
+  static const int OVERLAY_MARGIN = 2;
+
+  // Static overlay pen width
+  static const int OVERLAY_PEN_WIDTH = 5;
+
+protected:
+  void paintEvent(QPaintEvent *) {
+    QPainter p(this);
```

Comment:
shoudn't we have an implementation file for this class? (instead of those pseudo-inline methods)